src/mesh/boundary_standard.cxx

```diff
@@ -121,7 +125,13 @@ void BoundaryDirichlet::apply(Field2D &f){
 void BoundaryDirichlet::apply(Field2D &f,BoutReal t) {
   // Set (at 2nd order) the value at the mid-point between the guard cell and the grid cell to be val
   // N.B. Only first guard cells (closest to the grid) should ever be used
-  
+
+  Mesh* localmesh = f.getMesh();
+
+  // Check for staggered grids
+  CELL_LOC loc = f.getLocation();
+  ASSERT1(localmesh->StaggerGrids || loc == CELL_CENTRE);
```

Comment:
I think this can never fail -- it shouldn't be possible to set a location other than CELL_CENTRE if staggered grids are disabled. I guess it doesn't hurt much to have a check here, but our units tests should (hopefully) tell us if this condition is ever accidentally relaxed.

Reply:
True, I don't think this is an enlightening precondition, but as you say, it doesn't hurt.